### PR TITLE
First attempt at fixing PTV-436

### PR DESCRIPTION
**Pull request by jmchandonia**
A detailed list of changes is here:
https://kbase-jira.atlassian.net/browse/PTV-436?focusedCommentId=52407&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-52407

### CODE REVIEW

File: lib/SpeciesTreeBuilder/SpeciesTreeBuilderClient.pm [hunk 1]
```diff
@@ -81,20 +81,19 @@ sub new
...
-	    $self->{client}->{token} = $token->token;
-	}
-        else
-        {
-	    #
-	    # All methods in this module require authentication. In this case, if we
-	    # don't have a token, we can't continue.
-	    #
-	    die "Authentication failed: " . $token->error_message;
+	    $self->{client}->{token} = $self->{token};
```
Comment (sychan) [inline lib/SpeciesTreeBuilder/SpeciesTreeBuilderClient.pm]: You should handle the case of no token.

Comment (jmchandonia) [reply lib/SpeciesTreeBuilder/SpeciesTreeBuilderClient.pm]: That client code is auto-generated by the SDK.  I'm using SDK version 1.0.15 (commit 3471b17d8b69fa54ed44b6784262d1a2948c2283).  Is there something better generated by newer SDK versions that I should use instead?

**Comment by arfathpasha** [comment]
Looks good to me. Would be good to add a RELEASE_NOTES.md file with a changelog for this and future versions

**Comment by scanon** [comment]
@jmchandonia can you add a release notes?

**Comment by dcchivian** [comment]
@jmchandonia oh god please can we get this out the door!

**Comment by jmchandonia** [comment]
OK, release notes are added.

**Comment by arfathpasha** [comment]
@sychan PR is good to merge.

**Pull request merged**